prometheus/counter.go

```diff
@@ -136,7 +137,8 @@ func NewCounterVec(opts CounterOpts, labelNames []string) *CounterVec {
 	return &CounterVec{
 		metricVec: newMetricVec(desc, func(lvs ...string) Metric {
 			if len(lvs) != len(desc.variableLabels) {
-				panic(errInconsistentCardinality)
+				err := fmt.Errorf("%s: labels and values for '%s' don't match: labels %v, values %v", errInconsistentCardinality, desc.fqName, desc.variableLabels, lvs)
+				panic(err)
```

Comment:
You are not using the `err` variable anywhere else. Why not just `panic` directly without the `err` indirection? (See my previous comment for how that would look like in code.)